src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -573,7 +573,7 @@ public Boolean call() throws Exception
 
             // failed to get connection from db, sleep and retry
             quietlySleep(sleepBackoff);
-            sleepBackoff = Math.min(connectionTimeout / 2, (long) (sleepBackoff * 1.3));
+            sleepBackoff = Math.min(connectionTimeout / 5, (long) (sleepBackoff * 1.3));
```

Comment:
connection attempts per second is another angle.
Knowing connectiontimeout is 'usually' default or more, not 1second, which is more than 6 seconds of sleep for every failed attempt, HikariCP will be reasonably quicker to respond to network failure and fixes.